src/diffpy/labpdfproc/tests/test_functions.py

```diff
@@ -0,0 +1,65 @@
+import numpy as np
+import pytest
+
+from diffpy.labpdfproc.functions import Gridded_circle
+
+params1 = [
+    [[1, 3, 1, {(0.0, -1.0), (0.0, 0.0), (-1.0, 0.0), (1.0, 0.0), (0.0, 1.0)}, 5]],
+    [[0.2, 8, 1, {(-0.08571428571428572, -0.14285714285714285), (-0.02857142857142858, -0.14285714285714285), (0.02857142857142858, -0.14285714285714285), (0.08571428571428574, -0.14285714285714285), (-0.14285714285714285, -0.08571428571428572), (-0.08571428571428572, -0.08571428571428572), (-0.02857142857142858, -0.08571428571428572), (0.02857142857142858, -0.08571428571428572), (0.08571428571428574, -0.08571428571428572), (0.14285714285714285, -0.08571428571428572), (-0.14285714285714285, -0.02857142857142858), (-0.08571428571428572, -0.02857142857142858), (-0.02857142857142858, -0.02857142857142858), (0.02857142857142858, -0.02857142857142858), (0.08571428571428574, -0.02857142857142858), (0.14285714285714285, -0.02857142857142858), (-0.14285714285714285, 0.02857142857142858), (-0.08571428571428572, 0.02857142857142858), (-0.02857142857142858, 0.02857142857142858), (0.02857142857142858, 0.02857142857142858), (0.08571428571428574, 0.02857142857142858), (0.14285714285714285, 0.02857142857142858), (-0.14285714285714285, 0.08571428571428574), (-0.08571428571428572, 0.08571428571428574), (-0.02857142857142858, 0.08571428571428574), (0.02857142857142858, 0.08571428571428574), (0.08571428571428574, 0.08571428571428574), (0.14285714285714285, 0.08571428571428574), (-0.08571428571428572, 0.14285714285714285), (-0.02857142857142858, 0.14285714285714285), (0.02857142857142858, 0.14285714285714285), (0.08571428571428574, 0.14285714285714285)}, 32]]
+]
+
+@pytest.mark.parametrize('params1', params1)
+def test_get_grid_points(params1):
+    for param_set in params1:
+        radius, n_points_on_diameter, mu, expected_grid, expected_total_points = param_set
+        actual_gs = Gridded_circle(radius=radius, n_points_on_diameter=n_points_on_diameter, mu=mu)
+        assert actual_gs.grid == expected_grid
+        assert actual_gs.total_points_in_grid == expected_total_points
+
+
+
+# values of distances are correct but im a bit confused about the order of distances
+params2 = [
+    [[1, 3, 1, [25, 90, 140],
+     [[0, 2, 1.81261557406, 2, 0.845236523481399],
+      [0, 2, 0, 2, 2],
+      [0, 2, 0, 3.5320888862379562, 1.2855752193730785]]]],
```

Comment:
many of these numbers seem magical and too high in precision (is it important for your function that it gets a number correct to 11 significant figures?).  Can you manually compute a few special cases and use those as the test?  It may be harder here than for the grid points, so just find a reasonable balance, but don't use the function to compute these.